hack/generator/pkg/astbuilder/builder.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astbuilder
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+)
+
+// CheckErrorAndReturn checks if the err is non-nil, and if it is returns.
+func CheckErrorAndReturn(otherReturns ...*ast.Ident) ast.Stmt {
+
+	var returnValues []ast.Expr
+	for _, ret := range otherReturns {
+		returnValues = append(returnValues, ret)
+	}
+	returnValues = append(returnValues, ast.NewIdent("err"))
+
+	return &ast.IfStmt{
+		Cond: &ast.BinaryExpr{
+			X:  ast.NewIdent("err"),
```

Comment:
Is the assumption that it's always called `err` safe?

Reply:
It's _pretty_ safe and these are intended to be helpers for common patterns not replace the `ast` library entirely so I think as long as they cater for the 80+% case they've achieved their objective.

Adding support for other configuration increases the complexity of the signature and my objective was to keep the signatures as simple as I could (in contrast to ast which has quite complex signatures for most of the struct constructions because they need to support all the cases).

I think if/when we run into a situation where it can't be `err` (not sure what that would be off the top of my head) we can at that point either:
1. Just use raw `ast` for that uncommon case.
2. Update this method to allow an `errIdent` parameter and find/update callers.
3. Add a new method/override with more configuration options.

Reply:
Works for me.